main/test/flix/Test.Dec.Assoc.Type.Eff.flix

```diff
@@ -32,7 +32,8 @@ mod Test.Dec.Assoc.Type.Eff {
     instance Runner[Vector[a]] with Runner[a] {
         pub type E = Runner.E[a]
         pub def exec(x: Vector[a]): Unit \ Runner.E[a] =
-            foreach (a <- x) Runner.exec(a)
+            let f: a -> Unit \ Runner.E[a] = Runner.exec;
```

Comment:
Without this annotation, the compiler times out.

I think this is a bug that is uncovered here but not introduced. Justification: if i check out master and manually desugar this `foreach` loop into a `ForEach.forEach`, I get the same timeout. So it's not related to the actual `foreach` construct and it's not introduced in this PR.

Is this a known issue or should I make a ticket? I don't think it should block merging this.

Reply:
Interesting... You can make a ticket